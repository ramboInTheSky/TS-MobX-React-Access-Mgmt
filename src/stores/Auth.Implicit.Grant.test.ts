
import { AuthStore } from './Auth.Implicit.Grant'

jest.mock('auth0-js', () => ({
    WebAuth: jest.fn().mockImplementation(() => {
        return {
            parseHash: jest.fn(),
            client: {
                userInfo: jest.fn(),
            },
            authorize: jest.fn(),
            checkSession: jest.fn(),
            logout: jest.fn(),
        }
    }),
}))

describe('<AuthStore />', () => {
    it('renders correctly', () => {
        const store = new AuthStore()
        expect(store).toMatchSnapshot()
    })

    it('calls parseHash when attempting to retrieve a user session', () => {
        const store = new AuthStore()
        store.getUser()
        expect(store.auth0Instance.parseHash).toHaveBeenCalled()
    })

    it('calls auth0.login when attempting to login', () => {
        const store = new AuthStore()
        store.login()
        expect(store.auth0Instance.authorize).toHaveBeenCalled()
    })

    it('calls auth0.checkSession when attempting to checkSession', () => {
        const store = new AuthStore()
        store.checkSession()
        expect(store.auth0Instance.checkSession).toHaveBeenCalled()
    })

    it('calls auth0.logout when attempting to logout', () => {
        const store = new AuthStore()
        store.logout()
        expect(store.auth0Instance.logout).toHaveBeenCalled()
        expect(store.user).toBe(undefined)
    })
})
