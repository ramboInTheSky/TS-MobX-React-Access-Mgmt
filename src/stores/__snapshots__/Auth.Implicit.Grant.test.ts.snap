// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthStore /> renders correctly 1`] = `
AuthStore {
  "auth0": Object {
    "authorize": [MockFunction],
    "checkSession": [MockFunction],
    "client": Object {
      "userInfo": [MockFunction],
    },
    "logout": [MockFunction],
    "parseHash": [MockFunction],
  },
  "checkSession": [Function],
  "checkSessionInterval": undefined,
  "expiresAt": undefined,
  "getUser": [Function],
  "idToken": undefined,
  "loggedUser": undefined,
  "login": [Function],
  "logout": [Function],
  "logoutTimeout": null,
  "redirect": undefined,
  "settings": Object {
    "clientID": "",
    "domain": "",
    "redirectUri": "http://localhost",
    "responseType": "code",
  },
}
`;
